core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -86,10 +89,106 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
   /**
    * Get the amount of resources for the specified type from the resource list
    */
-  protected def getResource(res: List[Resource], name: String): Double = {
+  protected def getResource(res: JList[Resource], name: String): Double = {
     for (r <- res if r.getName == name) {
       return r.getScalar.getValue
     }
     0.0
   }
+
+
+  /** Helper method to get the key,value-set pair for a Mesos Attribute protobuf */
+  private[mesos] def getAttribute(attr: Attribute): (String, Set[String]) =
+    (attr.getName, attr.getText.getValue.split(',').toSet)
+
+
+  /** Build a Mesos resource protobuf object */
+  private[mesos] def createResource(resourceName: String, quantity: Double): Protos.Resource = {
+    Resource.newBuilder()
+      .setName(resourceName)
+      .setType(Value.Type.SCALAR)
+      .setScalar(Value.Scalar.newBuilder().setValue(quantity).build())
+      .build()
+  }
+
+
+  /**
+   * Match the requirements (if any) to the offer attributes.
+   * if attribute requirements are not specified - return true
+   * else if attribute is defined and no values are given, simple attribute presence is preformed
+   * else if attribute name and value is specified, subset match is performed on slave attributes
+   */
+  private[mesos] def matchesAttributeRequirements(
+      slaveOfferConstraints: Map[String, Set[String]],
+      offerAttributes: Map[String, Set[String]]): Boolean =
+    if (slaveOfferConstraints.isEmpty) {
+      true
+    } else {
+      slaveOfferConstraints.forall {
```

Comment:
I think dragos mentioned this for another case earlier -- I don't think you need the `if (slaveOfferConstraints.isEmpty)` since `forall` takes care of it for you
```scala
scala> List().forall{_ => 1 > 2}
res0: Boolean = true
```